hooks/pre-exit

```diff
@@ -20,10 +20,22 @@ readonly job_cleanup_retry_sec="${BUILDKITE_PLUGIN_K8S_JOB_CLEANUP_RETRY_INTERVA
 readonly job_cleanup_timeout_sec="${BUILDKITE_PLUGIN_K8S_JOB_CLEANUP_TIMEOUT_SEC:-60}"
 readonly job_cleanup_start_time="$SECONDS"
 
+external_secrets=${BUILDKITE_PLUGIN_K8S_EXTERNAL_SECRETS:-}
+external_secrets_list=${BUILDKITE_PLUGIN_K8S_EXTERNAL_SECRETS_0:-}
+
+
+
+# If External Secrets were used, clean them up
+if [[ -n $external_secrets || -n $external_secrets_list ]]; then
+  echo "Deleting ExternalSecrets"
+  kubectl delete ExternalSecret $job_name
```

Comment:
I'm wondering if we could use some form of binding + propagation policies to ensure the job + external secrets are always cleaned up together?

Reply:
Can you elaborate? That sounds like it could be beneficial, I want to make sure everything is cleaned up at the end and nothing persists between jobs.

Reply:
I don't think I want the Secret to be bound to the job, I have some ideas on future functionality in which the secret would need to be used independently of the job.

Reply:
Hmm, but then this wouldn't be useful for that use-case anyways, right?

Reply:
The Secret currently gets cleaned up in the pre-exit step, so it would still be available to anything else running until that stage.